prebid-server/developers/add-new-bidder-go.md

```diff
@@ -546,7 +546,56 @@ func (a *adapter) MakeRequests(request *openrtb.BidRequest, requestInfo *adapter
 
 If your bidding server supports multiple currencies, please be sure to pass through the `request.cur` field. If your bidding server only bids in a single currency, such as USD or EUR, that's fine. Prebid Server will convert your bid to the request currency if you include it in the bid response, otherwise we assume USD and conversion will not occur.
 
-Please ensure you forward the bid floor (`request.imp[].bidfloor`) and bid floor currency (`request.imp[].bidfloorcur`) values to your bidding server for enforcement. You'll soon have access to currency conversion helper methods if your endpoint only supports floors in a single currency.
+Please ensure you forward the bid floor (`request.imp[].bidfloor`) and bid floor currency (`request.imp[].bidfloorcur`) values to your bidding server for enforcement. You have access to the currency conversion helper method `ConvertCurrency` in case your endpoint only supports floors in a single currency.
+
+<details markdown="1">
+  <summary>Example: Currency conversion needed for impression bid floor.</summary>
+
+```go
+func (a *adapter) MakeRequests(request *openrtb.BidRequest, requestInfo *adapters.ExtraRequestInfo) (*adapters.RequestData, []error) {
+  var requests []*adapters.RequestData
+  var errors []error
+  
+  requestCopy := *request
+  for _, imp := range request.Imp {
+
+    // Check if imp comes with bid floor amount defined in a foreign currency
+    if imp.BidFloor > 0 && imp.BidFloorCur != "" && strings.ToUpper(imp.BidFloorCur) != "USD" {
+
+      // Convert to US dollars
+      convertedValue, err := reqInfo.ConvertCurrency(imp.BidFloor, imp.BidFloorCur, "USD")
+      if err != nil {
+        errors = append(errors, err)
+        continue
+      }
+
+      // Update after conversion. All imp elements inside request.Imp are shallow copies
+      // therefore, their non-pointer values are not shared memory and are safe to modify
+      // without risking a data race condition
+      imp.BidFloorCur = "USD"
+      imp.BidFloor = convertedValue
+    }
+
+    requestCopy.Imp = []openrtb.Imp{imp}
+
+    requestJSON, err := json.Marshal(request)
+    if err != nil {
+      errors = append(errors, err)
+      continue
+    }
+  
+    requestData := &adapters.RequestData{
+      Method: "POST",
+      Uri:    a.endpoint,
+      Body:   requestJSON,
+    }
+    requests = append(requests, requestData)
+  }
+  return requests, errors
+}
+```
+</details>
+<p></p>
```

Comment:
I think this example is confusing because it also includes imp splitting. Could you please remove the imp splitting aspect and keep it focused on just bid floor currency conversion?

Reply:
Agree, removed the imp splitting part and simplified a bit